src/lib/provider/Web3FunctionProxyProvider.ts

```diff
@@ -61,7 +70,11 @@ export class Web3FunctionProxyProvider {
 
       // Forward RPC call to internal provider
       try {
-        const result = await this._provider.send(method, params);
+        const provider = this._providers.get(
+          chainId
+        ) as ethers.providers.StaticJsonRpcProvider;
+
```

Comment:
We should probably have a graceful error handling in case of a not found chain id